plugins/test/runner.proto

```diff
@@ -59,13 +69,8 @@ message Input {
 
 message Expectation {
   // Headers expected in the output.
-  repeated Header has_header = 1;
-  // Headers expected to be missing. Values are ignored.
-  repeated Header no_header = 2;
-  // Generalized header matching against serialized "key: value" lines.
-  repeated StringMatcher headers = 3;
-
-  // Content accompanying an immediate response or body invocation result.
+  optional HeaderMatcher headers = 1;
+  // Content accompanying a body invocation result.
```

Comment:
I don't like that this unnecessarily introduces a breaking (backwards incompatible) proto change. People might already be using this API, let's not break it for no reason.

Also the most common test case now "stutters" with `headers { has_header: }` syntax. I get that it makes code factoring easier to have a reused HeaderMatcher proto, but it's not worth it IMO.

Mike's comment of "is this even necessary" rings true. All we really need to do in this PR is track an internal boolean saying "the plugin has sent immediate response" and then avoid firing subsequent callbacks.